### DISCUSSION TITLE
issue: Invoking handleSubmit inside an another function to resolve void Promise type error breaks form submit

### DISCUSSION THREAD

**Discussion by ethanniser** [react-hook-form/react-hook-form#10000] [Q&A] [answered]
### Version Number

7.43.2

### Codesandbox/Expo snack

https://codesandbox.io/s/epic-cloud-zmz5zn?file=/src/App.tsx

### Steps to reproduce

1. Go to https://codesandbox.io/s/epic-cloud-zmz5zn?file=/src/App.tsx
2. Open Console
3. Enter text in first input and click submit, verify expected behavior in console
4. Enter text in second input and click submit, see unexpected behavior

### Expected behaviour

The `<form>` element's `onSubmit` attribute takes a callback to a function of type `React.FormEventHandler<HTMLFormElement>` which returns `void`, however `handleSubmit` from `useForm` returns `Promise<void>`

Hovering `onSubmit` attribute
![image](https://user-images.githubusercontent.com/100045248/221049724-f4735b31-b223-4e42-a26d-c2764d45b85f.png)

"Go to definiton" of the attribute
![image](https://user-images.githubusercontent.com/100045248/221049803-f1ecefd0-241a-41df-a94d-a4f7978402fb.png)

Hovering `onSubmit` callback
![image](https://user-images.githubusercontent.com/100045248/221049171-6757daab-1f2a-404b-95e2-cb7b02286f54.png)

This triggers ESlint rule ["no-misused-promises"](https://typescript-eslint.io/rules/no-misused-promises/) (I can't figure out how to get eslint working in CodeSandbox example, so this is not seen there.)

Placing the call to `handleSubmit` inside of another function that explicitly returns `void` and using that as the callback resolves the type error, but breaks react-hook-form. 

![image](https://user-images.githubusercontent.com/100045248/221051955-aa859b6f-42e8-4022-9796-b3af4f617a8d.png)

On the code sandbox the console log will go through for a flash, and then the whole page just reloads. However, when recreating the issue on my local machine what would happen when clicking submit is the default submitting behavior: a GET request sent to the current url with the form contents as URL Search Params, and the console log never running.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment, marked as answer]
#8020

**Comment by ethanniser** [comment]
Thank you for pointing me to this, my apologies for making a duplicate issue. I didn't find this when looking for a solution before posting.
